src/server/cluster/cluster_family.h

```diff
@@ -85,8 +85,15 @@ class ClusterFamily {
       ABSL_LOCKS_EXCLUDED(migration_mu_);
 
   void StartSlotMigrations(std::vector<MigrationInfo> migrations);
-  SlotRanges RemoveOutgoingMigrations(std::shared_ptr<ClusterConfig> new_config,
-                                      std::shared_ptr<ClusterConfig> old_config)
+
+  // must be destroyed excluded set_config_mu and migration_mu_ locks
```

Comment:
you could add such `DCHECK()`s in a custom d'tor

Reply:
perhaps you could even add that as `ABSL_LOCKS_EXCLUDED` to the d'tor decl, I imagine it would work